TransTest.hs

```diff
@@ -18,61 +18,50 @@ testDotGOutput = do
     putStrLn "=== testDotGOutput"
     assertEq threeSignalsConcept expected
   where
-    expected = ".model out\n"          ++
-               ".inputs A\n"           ++
-               ".outputs B\n"          ++
-               ".internal C\n"         ++
-               ".graph\n"              ++
-               "A0 A+\n"               ++
-               "A+ A1\n"               ++
-               "A1 A-\n"               ++
-               "A- A0\n"               ++
-               "B0 B+\n"               ++
-               "B+ B1\n"               ++
-               "B1 B-\n"               ++
-               "B- B0\n"               ++
-               "C0 C+\n"               ++
-               "C+ C1\n"               ++
-               "C1 C-\n"               ++
-               "C- C0\n"               ++
-               ".marking {A0 B1 C0}\n" ++
-               ".end\n"
+    threeSignalsConcept = doTranslate signs (inputs [a] <> outputs [b] <> internals [c] <> initialise0 [a, c] <> initialise1 [b])
+    a = Signal 0
+    b = Signal 1
+    c = Signal 2
+    signs = [a, b, c]
+    expected = unlines ([".model out", ".inputs A", ".outputs B", ".internal C", ".graph"] ++
+                        ["A0 A+", "A+ A1", "A1 A-", "A- A0", "B0 B+", "B+ B1", "B1 B-", "B- B0"] ++
+                        ["C0 C+", "C+ C1", "C1 C-", "C- C0", ".marking {A0 B1 C0}", ".end"])
 
 testHandshakeConcept :: IO ()
 testHandshakeConcept = do
     putStrLn "===testHandshakeConcept"
-    assertEq (sort handshakeConcept) (sort expected)
+    assertEq (sort (arcs handshakeConcept)) (sort expected)
   where
+    handshakeConcept = inputs[a] <> outputs [b] <> handshake00 a b
     expected = [([rise a], rise b), ([rise b], fall a), ([fall a], fall b), ([fall b], rise a)]
-    a = Signal 0
-    b = Signal 1
+    [a, b] = map Signal [0 .. 1]
 
 testOrGateConcept :: IO ()
 testOrGateConcept = do
     putStrLn "===testOrGateConcept"
-    assertEq (sort orGateConcept) (sort expected)
+    assertEq (sort (arcs orGateConcept)) (sort expected)
   where
+    orGateConcept = inputs [a, b] <> outputs [c] <> initialise0 [a, b, c] <> orGate a b c
     expected = [([rise a, rise b], rise c), ([fall a], fall c), ([fall b], fall c)]
-    a = Signal 0
-    b = Signal 1
-    c = Signal 2
+    [a, b, c] = map Signal [0..2]
 
 testTwoAndGates :: IO ()
 testTwoAndGates = do
     putStrLn "===testTwoAndGates"
-    assertEq (sort twoAndGatesConcept) (sort expected)
+    assertEq (sort (arcs twoAndGatesConcept)) (sort expected)
   where
+    twoAndGatesConcept = inputs [a, b, c, d] <> outputs [out] <> initialise0 [a, b, c, d, out] <> andGate a b out <> andGate c d out
     expected = [([rise a], rise out), ([rise b], rise out), ([rise c], rise out), ([rise d], rise out), ([fall a, fall b], fall out), ([fall c, fall d], fall out)]
-    a   = Signal 0
-    b   = Signal 1
-    c   = Signal 2
-    d   = Signal 3
-    out = Signal 4
+    [a, b, c, d, out] = map Signal [0..4]
```

Comment:
Perhaps, this could be a top-level declaration, then you don't need to declare signals in every test?